services/search/pkg/service/v0/service.go

```diff
@@ -3,85 +3,94 @@ package service
 import (
 	"context"
 	"errors"
-	"path/filepath"
-
-	"github.com/blevesearch/bleve/v2"
+	"fmt"
+	gateway "github.com/cs3org/go-cs3apis/cs3/gateway/v1beta1"
 	revactx "github.com/cs3org/reva/v2/pkg/ctx"
 	"github.com/cs3org/reva/v2/pkg/errtypes"
 	"github.com/cs3org/reva/v2/pkg/events"
 	"github.com/cs3org/reva/v2/pkg/events/server"
 	"github.com/cs3org/reva/v2/pkg/rgrpc/todo/pool"
 	"github.com/go-micro/plugins/v4/events/natsjs"
+	"github.com/owncloud/ocis/v2/services/search/pkg/content"
+	"github.com/owncloud/ocis/v2/services/search/pkg/engine"
+	"github.com/owncloud/ocis/v2/services/search/pkg/search"
 	merrors "go-micro.dev/v4/errors"
 	"go-micro.dev/v4/metadata"
 	grpcmetadata "google.golang.org/grpc/metadata"
 
 	"github.com/owncloud/ocis/v2/ocis-pkg/log"
 	searchsvc "github.com/owncloud/ocis/v2/protogen/gen/ocis/services/search/v0"
-	"github.com/owncloud/ocis/v2/services/search/pkg/config"
-	"github.com/owncloud/ocis/v2/services/search/pkg/search"
-	"github.com/owncloud/ocis/v2/services/search/pkg/search/index"
-	searchprovider "github.com/owncloud/ocis/v2/services/search/pkg/search/provider"
 )
 
 // NewHandler returns a service implementation for Service.
 func NewHandler(opts ...Option) (searchsvc.SearchProviderHandler, error) {
+	var eng engine.Engine
+	var gw gateway.GatewayAPIClient
+	var bus events.Stream
+	var extractor content.Extractor
+	var err error
```

Comment:
I personally find it clearer if vars are declared short before they are overwritten. for example:
```golang
var eng engine.Engine
switch cfg.Engine.Type {
    // assign to eng
}
```

But if you want to define multiple at once please use multi-assign notation:
```golang
var (
    eng engine.Engine
    gw  gateway.GatewayAPIClient
)
```